frame/identity/src/lib.rs

```diff
@@ -862,14 +864,59 @@ decl_module! {
 	}
 }
 
+impl<T: Trait> Benchmarking<BenchmarkResults> for Module<T> {
+	const STEPS: u32 = 100;
+	const REPEATS: u32 = 10;
+
+	fn run_benchmarks() -> Vec<BenchmarkResults> {
+		// first one is set_identity.
+		let components = benchmarking::set_identity::components();
+		// results go here
+		let mut results: Vec<BenchmarkResults> = Vec::new();
+		// Select the component we will be benchmarking. Each component will be benchmarked.
+		for (name, low, high) in components.iter() {
+			// Create up to `STEPS` steps for that component between high and low.
+			let step_size = ((high - low) / Self::STEPS).max(1);
+			let num_of_steps = (high - low) / step_size;
+			for s in 0..num_of_steps {
+				// This is the value we will be testing for component `name`
+				let component_value = step_size * s;
+
+				// Select the mid value for all the other components.
+				let c: Vec<(BenchmarkParameter, u32)> = components.iter()
+					.map(|(n, l, h)|
+						(*n, if n == name { component_value } else { (h - l) / 2 + l })
+					).collect();
+
+				for _r in 0..Self::REPEATS {
+					let instance = benchmarking::set_identity::instance(&c);
+					let start = sp_io::offchain::timestamp();
+					instance.dispatch(Some(0).into());
+					let finish = sp_io::offchain::timestamp();
+					let elapsed = finish.0 - start.0;
+					results.push((c.clone(), elapsed));
+				}
+			}
+		}
+		return results;
+	}
+}
+
+sp_api::decl_runtime_apis! {
```

Comment:
I think we usually place the apis outside of the runtime? but maybe it is not a big deal.

Reply:
Indeed I think at the end we should have only one runtime_api function which is `run_benchmarks(module_to_run: Vec<ModuleInde>)` or something like that

Reply:
I was working on this and it should be placed in a std sub-crate of the frame module. See how system is defining some extra runtime api for its rpc.